textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/relation/RelationManager.java

```diff
@@ -88,6 +88,11 @@ public void createTable(String tableName, String indexDirectory, Schema schema,
 
     // drop a table
     public void deleteTable(String tableName) throws StorageException {
+        if (tableName.equals(CatalogConstants.TABLE_CATALOG) 
```

Comment:
Add a comment to the codebase to explain the logic: "Users cannot delete catalog tables."